### ISSUE TITLE
X11/XLib keystrokes registered "akin to a text editor"

### ISSUE THREAD

**Issue by HitCoder9768** [enigma-dev/enigma-dev#1375]
as presented on https://enigma-dev.org/forums/index.php?topic=2922 user tried making game on linux, using keyboard inputs.
X11/XLib currently has problems that are not present on SDL within the same OS, where holding down a key will at first register the key as being held down, and after a timeout, will register a spam-input of the key similar to that of holding down a key while in a text field.
This causes problems for keyboard-input-to-player-movement as a player object will start to stutter rather than moving smoothly, in addition to other key inputs being subject to latency.

**Comment by RobertBColton** [comment]
Alright, I did the actual bisect and traced the input lag to 15fce1a894de6c2f5ec1b4443fd1cdb5a02dbf9c which generalized the main loop. Something in that commit/pr introduced the input lag. The input lag does NOT occur in d0d0ecb4fdcee7c75a55dcf0522e0c3c3d92edfc which was the last commit to master just before the generalization of main in which I tweaked the highscore functions a little for Hugar.
